### ISSUE TITLE
Chrome on iOS, download instead of AR

### ISSUE THREAD

**Issue by MajorMeerkatThe3rd** [google/model-viewer#3235]
### Description

We have a problem with AR, specifically quick look in Chrome. As we want to integrate the model-viewer into our automatic processes, we need to store the model on our servers. Unfortunately, in Chrome on iOS, this does not seem to work. We tried multiple approaches, but somehow Chrome still want to download the file, and not display it in AR.

#### Live Demo
Here the Astronaut from your site:
[Working Example](https://glitch.com/edit/#!/comfortable-torpid-collar)

Here the Astronaut from your site, uploaded to our servers:
[Not Working Example](https://glitch.com/edit/#!/pointy-bronze-denim)

### Version
- model-viewer: v1.10.1

### Browser Affected
- [ X ] Chrome, version: 97.0.4692.84
- [ ] Edge
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS
- [ ] Android
- [ X ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR
- [ ] WebXR
- [ ] SceneViewer
- [ X ] QuickLook

**Comment by samaneh-kazemi** [comment]
Hi @MajorMeerkatThe3rd, thanks for reaching out. This is a known chrome/ios issue. similar to #2449 
Chrome tracking bug:

https://bugs.chromium.org/p/chromium/issues/detail?id=1197345

**Comment by elalish** [comment]
Actually, looking at your Glitch, I wonder if it's just that the file type is unknown. The only difference I see with your `ios-src` URL vs. ours is that yours lacks the `.usdz` file extension. Does it make a difference if you change that?

**Comment by rawwerks** [comment]
I think this is the same bug, so commenting here instead of making a new one: 

**TL;DR - AR does not launch in Google Chrome 99.0.4844.59 on iOS 15.3.1**

Model-Viewer still returns .canActivateAR, however.

https://ar.ink did not successfully load world peace in AR, so I immediately went to https://modelviewer.dev , which doesn't work either. The AR icon just disappears without launching the AR experience.

https://user-images.githubusercontent.com/19483938/157382542-d2dfc227-c25c-40dd-8e34-b5e6f5f61f46.mp4

One would naively assume that the Google browser would run the Google model-viewer...so maybe this is an iOS issue?

(Firefox doesn't work either, but I'll save that for another issue.)

**Comment by elalish** [comment]
Yeah, Firefox doesn't support WebXR and Chrome iOS has a known bug where auto-generated USDZs don't work. It should work if use a dedicated ios-src: https://modelviewer.dev/examples/augmentedreality/#ar

**Comment by rawwerks** [comment]
@elalish - is this something that you could detect and pass to .canActivateAR ? right now i'm relying on .canActivateAR as the UX switch (which i believe is it's intended purpose) - but it's not catching either the Firefox or Chrome iOS issues.

if this is not feasible - is there some example code or known best practice for doing device/browser logic to avoid sending the user down a dead end? i understand that AR adds a lot of complexity to building a 'responsive' website. right now it feels like a bit of a free-for-all where everyone is on their own...but this is not my field, so perhaps i'm not aware of the best practices.

**Comment by MajorMeerkatThe3rd** [comment]
Hey guys, thanks for your help. I got in contact with the Chromium team and this is an issue on their side. 
see here: https://bugs.chromium.org/p/chromium/issues/detail?id=1302551
Summarized, either you have to supply an url-ending or the Content-Disposition header (or both). The Content-Type is not recognized at the moment, which will be fixed in version 100.
Cheers

**Comment by elalish** [comment]
@rawwerks Interesting, would you mind debugging a little to see where the error is? I'd guess this is returning true when it shouldn't: https://github.com/google/model-viewer/blob/067e36b5a2311f618286ada8b16794e675f54b70/packages/model-viewer/src/constants.ts#L64

If that's the case, it's an Apple bug, since they are declaring support in their WebView without actually implementing it, in which case you should file a bug on WebKit. If you find the right combination of userAgent strings to check, you're also welcome to put in a PR to work around the issue.

**Comment by sprengerst** [comment]
Hi, it's still an issue even with Chrome 100. 
The usdz generated by prepareUsdz() doesn't seem to start quicklook.

If we start with Safari everything works as it should.

Best,
Stefan

**Comment by amcc** [comment]
in case its useful to this thread - i'm having an issue with the astronaut on the home page of modelviewer - but only that particular model. I get the same symptoms as above in chrome on ios and in safari, i.e. the AR button just disappears when i click it. Other models such as the simple astronaut in examples work fine in safari - and download as a file in chrome (appear as text on the browser). So is this a combined issue - one with that actual file and one with ios chrome itself?

**Comment by elalish** [comment]
This has been fixed in Chrome 102.

**Comment by vseijas** [comment]
I'm having this issue for older iPads on chrome (the glb file gets downloaded) and for iPhone 12 and older on chrome: the quick view button disappears when clicked. Not allowing AR experience

**Comment by elalish** [comment]
I think this was dependent on a newish Apple API, so if either the Chrome or iOS version is old, it's unlikely this will work. Not much we can do - this is life on the bleeding edge in AR.
